code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1289,109 +1274,94 @@
 	else
 		to_chat(usr, "<span class='notice'>[self ? "Your" : "[src]'s"] pulse is [src.get_pulse(GETPULSE_HAND)].</span>")
 
-/mob/living/carbon/human/proc/set_species(var/new_species, var/default_colour, var/delay_icon_update = 0)
-	var/datum/species/oldspecies = species
-	var/datum/species/NS = all_species[new_species]
-	if(!dna)
-		if(!new_species)
-			new_species = "Human"
-	else
-		if(!new_species)
-			new_species = dna.species
-		else
-			dna.species = new_species
+/mob/living/carbon/human/proc/set_species(datum/species/new_species, default_colour, delay_icon_update = FALSE)
+	var/datum/species/oldspecies = dna.species
 
-	if(species)
-		if(species.name && species.name == new_species)
-			return
+	if(oldspecies)
+		if(oldspecies.language)
+			remove_language(oldspecies.language)
 
-		if(species.language)
-			remove_language(species.language)
+		if(oldspecies.default_language)
+			remove_language(oldspecies.default_language)
 
-		if(species.default_language)
-			remove_language(species.default_language)
+		if(gender == PLURAL && oldspecies.has_gender)
+			change_gender(pick(MALE, FEMALE))
 
-		if(gender == PLURAL && NS.has_gender)
-			change_gender(pick(MALE,FEMALE))
-		species.handle_pre_change(src)
+		if(oldspecies.default_genes.len)
+			oldspecies.handle_dna(src, TRUE) // Remove any genes that belong to the old species
 
-	species = all_species[new_species]
+		oldspecies.handle_pre_change(src)
 
-	if(oldspecies)
-		if(oldspecies.default_genes.len)
-			oldspecies.handle_dna(src,1) // Remove any genes that belong to the old species
+	dna.species = new new_species()
 
-	tail = species.tail
+	tail = dna.species.tail
 
-	maxHealth = species.total_health
+	maxHealth = dna.species.total_health
 
-	if(species.language)
-		add_language(species.language)
+	if(dna.species.language)
+		add_language(dna.species.language)
 
-	if(species.default_language)
-		add_language(species.default_language)
+	if(dna.species.default_language)
+		add_language(dna.species.default_language)
 
-	hunger_drain = species.hunger_drain
-	digestion_ratio = species.digestion_ratio
+	hunger_drain = dna.species.hunger_drain
+	digestion_ratio = dna.species.digestion_ratio
 
-	if(species.base_color && default_colour)
+	if(dna.species.base_color && default_colour)
 		//Apply colour.
-		skin_colour = species.base_color
+		skin_colour = dna.species.base_color
 	else
 		skin_colour = "#000000"
 
-	if(!(species.bodyflags & HAS_SKIN_TONE))
+	if(!(dna.species.bodyflags & HAS_SKIN_TONE))
 		s_tone = 0
 
-	species.create_organs(src)
+	dna.species.create_organs(src)

```

Comment:
@Fox-McCloud The organ regeneration would be a consequence of this line here. I'm not sure how to rectify it, since it'd be difficult to tell which organs on the old species map to which organs on the new species

Reply:
I added a solution for this that returns if the species is the same as it previously was; there's an override though, or else humans, at round start, will spawn in, then immediately die and have no icons (since default DNA has a species human datum for safety).

Reply:
pfff